platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformBuilder.java

```diff
@@ -0,0 +1,375 @@
+            @NonNull final String swirldName,
+            @NonNull final SoftwareVersion softwareVersion,
+            @NonNull final Supplier<SwirldState> genesisStateBuilder,
+            @NonNull final NodeId selfId) {
+
+        this.appName = Objects.requireNonNull(appName);
+        this.swirldName = Objects.requireNonNull(swirldName);
+        this.softwareVersion = Objects.requireNonNull(softwareVersion);
+        this.genesisStateBuilder = Objects.requireNonNull(genesisStateBuilder);
+        this.selfId = Objects.requireNonNull(selfId);
+    }
+
+    /**
+     * Set the configuration builder to use. If not provided then one is generated when the platform is built.
+     *
+     * @param configurationBuilder the configuration builder to use
+     * @return this
+     */
+    @NonNull
+    public PlatformBuilder withConfigurationBuilder(@Nullable final ConfigurationBuilder configurationBuilder) {
+        this.configurationBuilder = configurationBuilder;
+        return this;
+    }
+
+    /**
+     * The path to the settings file. Default is "./settings.txt" where "./" is the current working directory of the
+     * JVM.
+     *
+     * @param path the path to settings.txt
+     * @return this
+     * @throws IllegalArgumentException if the path does not exist
+     */
+    public PlatformBuilder withSettingsPath(@NonNull final Path path) {
+        Objects.requireNonNull(path);
+        final Path absolutePath = getAbsolutePath(path);
+        if (!Files.exists(absolutePath)) {
+            throw new IllegalArgumentException("File " + absolutePath + " does not exist");
+        }
+        this.settingsPath = absolutePath;
+        return this;
+    }
+
+    /**
+     * The path to the config file. Default is "./config.txt" where "./" is the current working directory of the JVM.
+     *
+     * @param path the path to config.txt
+     * @return this
+     * @throws IllegalArgumentException if the path does not exist
+     */
+    public PlatformBuilder withConfigPath(@NonNull final Path path) {
+        Objects.requireNonNull(path);
+        final Path absolutePath = getAbsolutePath(path);
+        if (!Files.exists(absolutePath)) {
+            throw new IllegalArgumentException("File " + absolutePath + " does not exist");
+        }
+        this.configPath = absolutePath;
+        return this;
+    }
+
+    /**
+     * Build a platform but do not start it.
+     *
+     * @return a new platform instance
+     */
+    public SwirldsPlatform build() {
+        return build(false);
+    }
+
+    /**
+     * Setup static utilities. If running multiple platforms in the same JVM, this should be called when constructing
+     * the first platform only.
+     *
+     * @param configuration the configuration for this node
+     * @return true if this is the first time this method has been called, false otherwise
+     */
+    private boolean setupStaticUtilities(@NonNull final Configuration configuration) {
```

Comment:
I ask my self if we can externalize that to have the builder not use static. Maybe a `StaticPlatformBuilder` as a singleton that is just called once by this builder class.

Reply:
I've moved the static stuff to a helper class.